substrate/network-libp2p/src/network_state.rs

```diff
@@ -691,17 +695,68 @@ fn is_peer_disabled(
 	}
 }
 
-/// Parses an address of the form `/ip4/x.x.x.x/tcp/x/p2p/xxxxxx`, and adds it
-/// to the given peerstore. Returns the corresponding peer ID.
+/// Parses an address of the form `/ip4/x.x.x.x/tcp/PORT/p2p/NODE_ID` or
+/// `[enode://|p2p://]NODE_ID@IP:PORT`, and adds it to the given peerstore.
```

Comment:
this seems like it could cause confusion. `enode://` addresses are for devp2p hosts. (and i've never seem `p2p://` addresses.) the two formats i'd like to support are multiaddr `/ip4/<IP>/tcp/<PORT>/p2p/<MULTIHASH_NODE_ID>` and `<IP>:<PORT>` (letting the p2p subsystem figure out the `NODE_ID`.